File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        ListNode tail = list2;
        while(tail.next != null) tail = tail.next;

        ListNode prev = null, curr = list1;
        for(int i = 0; i<a; i++) {
            prev = curr;
            curr = curr.next;
        }

        prev.next = list2;
        for(int i = a; i<=b; i++) curr = curr.next;

        tail.next = curr;
        return list1;
    }
}

File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def mergeInBetween(self, list1, a, b, list2):
        tail = list2
        while tail.next:
            tail = tail.next
        
        prev, curr = None, list1
        for i in range(a): 
            prev = curr
            curr = curr.next
        
        prev.next = list2
        for i in range(a, b + 1):
            curr = curr.next
        
        tail.next = curr
        return list1
        